packages/consumption/src/modules/requests/incoming/IncomingRequestsController.ts

```diff
@@ -245,6 +272,150 @@ export class IncomingRequestsController extends ConsumptionBaseController {
         }
     }
 
+    private validateKeyUniquenessOfRelationshipAttributesWithinRequest(
+        items: (RequestItem | RequestItemGroup)[],
+        params: (DecideRequestItemParametersJSON | DecideRequestItemGroupParametersJSON)[]
+    ) {
+        const fragmentsOfMustBeAcceptedItemsOfRequest: RelationshipAttributeFragment[] = [];
+        const fragmentsOfAcceptedItemsOfRequest: RelationshipAttributeFragment[] = [];
+
+        for (let i = 0; i < params.length; i++) {
+            const item = items[i];
+            const decideItemParams = params[i];
+
+            if (item instanceof RequestItemGroup) {
+                const fragmentsOfMustBeAcceptedItemsOfGroup = this.extractRelationshipAttributeFragmentsFromMustBeAcceptedItemsOfGroup(item);
+                if (fragmentsOfMustBeAcceptedItemsOfGroup) {
+                    fragmentsOfMustBeAcceptedItemsOfRequest.push(...fragmentsOfMustBeAcceptedItemsOfGroup);
+                }
+
+                const fragmentsOfAcceptedItemsOfGroup = this.extractRelationshipAttributeFragmentsFromAcceptedItemsOfGroup(
+                    item,
+                    decideItemParams as DecideRequestItemGroupParametersJSON
+                );
+                if (fragmentsOfAcceptedItemsOfGroup) {
+                    fragmentsOfAcceptedItemsOfRequest.push(...fragmentsOfAcceptedItemsOfGroup);
+                }
+            } else {
+                const fragmentOfMustBeAcceptedRequestItem = this.extractRelationshipAttributeFragmentFromMustBeAcceptedRequestItem(item);
+                if (fragmentOfMustBeAcceptedRequestItem) {
+                    fragmentsOfMustBeAcceptedItemsOfRequest.push(fragmentOfMustBeAcceptedRequestItem);
+                }
+
+                const fragmentOfAcceptedRequestItem = this.extractRelationshipAttributeFragmentFromAcceptedRequestItem(item, decideItemParams as DecideRequestItemParametersJSON);
+                if (fragmentOfAcceptedRequestItem) {
+                    fragmentsOfAcceptedItemsOfRequest.push(fragmentOfAcceptedRequestItem);
+                }
+            }
+        }
+
+        if (IncomingRequestsController.containsDuplicateRelationshipAttributeFragments(fragmentsOfMustBeAcceptedItemsOfRequest)) {
+            throw ConsumptionCoreErrors.requests.violatedKeyUniquenessOfRelationshipAttributes(
+                "The Request can never be accepted because it would lead to the creation of more than one RelationshipAttribute in the context of this Relationship with the same key."
+            );
+        } else if (IncomingRequestsController.containsDuplicateRelationshipAttributeFragments(fragmentsOfAcceptedItemsOfRequest)) {
+            return ValidationResult.error(
+                ConsumptionCoreErrors.requests.invalidAcceptParameters(
+                    "The Request cannot be accepted with this parameters because it would lead to the creation of more than one RelationshipAttribute in the context of this Relationship with the same key."
```

Comment:
It would be good if the problematic requestItems were named here (e. g. the way it happens in requestItemProcessors)

Reply:
How exactly could this be achieved? Since the problematic RequestItems cannot be identified by an ID, I find it difficult not to remain vague.

Reply:
Just the key will have to do then